zerver/forms.py

```diff
@@ -154,8 +145,8 @@ def clean_email(self):
             return email
 
         # If no realm is specified, fail
-        realm = get_valid_realm(email)
-        if realm is None:
+        realm = get_realm(resolve_email_to_domain(email))
```

Comment:
Both of these are done in subsequent commits; should i move the changes earlier?